### PR TITLE
feat(plugin-observability): add telemetry notice

### LINKED ISSUE #5374: [composer] show telemetry collection warning
### Motivation

We want to respect people's privacy while gathering what composer telemetry we can.

### Finish Line

- [ ] when starting up composer for the first time, let people opt out of telemetry
- [ ] add a setting to the Settings panel that indicates whether telemetry is enabled and allows the user to toggle it on/off
- [ ] document the telemetry we're collecting in our docs

### PR DESCRIPTION
**Pull request by wittjosiah**
Resolves #5374

Screenshot 2024-02-02 at 12 30 38

### CODE REVIEW

File: packages/apps/plugins/plugin-layout/src/components/Toast.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  description,
+  icon,
+  duration,
+  actionLabel,
+  actionAlt,
+  closeLabel,
+  onAction,
+}: ToastProps) => {
+  return (
+    <NaturalToast.Root defaultOpen duration={duration}>
```
Comment (thure) [inline packages/apps/plugins/plugin-layout/src/components/Toast.tsx:21]: I was wondering if the `Natural{Component}` naming would stick around 😅

File: packages/sdk/app-framework/src/plugins/common/layout.ts [hunk 2]
```diff
@@ -11,22 +11,42 @@ import type { Plugin } from '../PluginHost';
 // Provides
 //
 
+export const Toast = z.object({
+  id: z.string(),
+  title: z.string().optional(),
+  description: z.string().optional(),
+  // TODO(wittjosiah): `icon` should be string to be parsed by an `Icon` component.
```
Comment (thure) [inline packages/sdk/app-framework/src/plugins/common/layout.ts:18]: 💯, looking forward to making this happen.

**Review by thure — APPROVED** [review]
lgtm ✨

File: packages/apps/plugins/plugin-layout/src/components/Toast.tsx (hunk 1 above)
Comment (wittjosiah) [reply packages/apps/plugins/plugin-layout/src/components/Toast.tsx:21]: It's not my favourite, but I don't have a better alternative currently 😅

**Review by thure — APPROVED** [review]
lgtm!

**Pull request merged**